text/0240-app-telemetry-schema-vision-roadmap.md

```diff
@@ -0,0 +1,384 @@
+> was designed and deployed by Facebook to address the same type of problem but in
+> a proprietary context (refer to
+> this [positional paper](https://research.facebook.com/publications/positional-paper-schema-first-application-telemetry/)
+> for more information).
+
+### Development Strategies
+
+Two development strategies coexist and must be supported. The first strategy, a
+monorepo type (single source of truth), has complete control over the
+applications, their dependencies, and the associated telemetry schemas. The
+second strategy is more heterogeneous, composed of multiple repositories, where
+the build process of external dependencies is out of the control of the entity
+owning the applications or services.
+
+In the first model, each build process can independently apply telemetry schema
+policies in parallel, knowing that the policies are shared and the entire
+environment is controlled by the same entity.
+
+In the second model, the application or service environment does not control the
+telemetry schema policies of external dependencies. There is a need for a method
+to retrieve telemetry schemas from these dependencies. The mechanism for
+distributing these schemas is still being discussed (refer to the Open Questions
+section). Ultimately, this will enable the CI/CD pipeline of the application or
+service to apply its own local policies to these schemas from the dependencies.
+
+![Development Strategies to Support](./img/0240-otel-weaver-dev-strategies.svg)
+
+## Open Questions
+
+During the review of this document, several questions were raised, and some
+remain unresolved. We've decided to postpone the answers to these open questions
+to future OTEPs. This approach seems more practical as the resolution of these
+questions is not critical at this stage and will likely become clearer as we
+define and implement future OTEPs.
+
+### Should we use a single Telemetry Schema or the combination of Component and Resolved Telemetry Schema?
+
+The debate between adopting a single Telemetry Schema or separate Component and
+Resolved Telemetry Schemas remains unresolved.
+
+Advocates of a single-schema approach see it as a simplification in terms of
+schema definition, implementation, and even cognitive overhead. They suggest
+using the same schema but disallowing any external references for schemas
+intended for publication and reuse by third parties. The schema resolution
+process would then remove these references.
+
+Proponents of a two-schema approach believe that each format is intended for
+different users and use cases (app/lib developers vs telemetry tool developers),
+and therefore, having two distinct structures and formats would make it easier
+to optimize each for its specific use case (in multiple dimensions). The first
+group, app and lib developers, is much larger than the second and would not need
+to understand the details of the second format.
+
+### What should be the schema(s) be named?
```

Comment:
This is misleading, you can be asking about conceptual names (component vs. resolved schemas), or how they are implemented (same DSL or different)

Reply:
I have updated this sentence as follows: 
> Furthermore, the group of developers using the Component Telemetry Schema will most likely be much larger than the group of developers who need to understand the details of the Resolved Telemetry Schema. 

I hope it's less misleading.